catalog/mgmp/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/mgmp/transformer/MgmpTransformer.java

```diff
@@ -0,0 +1,594 @@
+                true);
+
+        List<String> types = pathValueTracker.getAllValues(toPath(GmdConstants.CRS_CODE_PATH));
+        List<String> codes = pathValueTracker.getAllValues(toPath(GmdConstants.CRS_AUTHORITY_PATH));
+        if (CollectionUtils.isNotEmpty(codes) && CollectionUtils.isNotEmpty(types)) {
+            for (int i = 0; i < codes.size(); i++) {
+                crsCodes.add(codes.get(i) + ":" + types.get(i));
+            }
+        }
+
+        if (CollectionUtils.isNotEmpty(mgmpTypes) && CollectionUtils.isNotEmpty(mgmpCodes)) {
+            for (int i = 0; i < mgmpCodes.size(); i++) {
+                crsCodes.add(mgmpTypes.get(i) + ":" + mgmpCodes.get(i));
+            }
+        }
+        metacard.setAttribute(Location.COORDINATE_REFERENCE_SYSTEM_CODE, (Serializable) crsCodes);
+    }
+
+    private void addMetacardIsrInformation(final XstreamPathValueTracker pathValueTracker,
+            MetacardImpl metacard) {
+        List<String> isrComments = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_COVERAGE_COMMENT_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_IMAGE_COMMENT_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_MD_IMAGE_COMMENT_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrComments,
+                MgmpConstants.ISR_VIDEO_COMMENT_PATH,
+                false);
+
+        List<String> isrCategory = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_COVERAGE_CATEGORY_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_IMAGE_DESCRIPTION_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_MD_IMAGE_DESCRIPTION_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                isrCategory,
+                MgmpConstants.ISR_VIDEO_DESCRIPTION_PATH,
+                false);
+
+        metacard.setAttribute(Isr.COMMENTS, (Serializable) isrComments);
+        metacard.setAttribute(Isr.CATEGORY, (Serializable) isrCategory);
+
+        addMetacardIsrOrganizationalUnit(metacard);
+    }
+
+    public void addMetacardIsrOrganizationalUnit(MetacardImpl metacard) {
+        Attribute pocName = metacard.getAttribute(Contact.POINT_OF_CONTACT_NAME);
+        Attribute publisherName = metacard.getAttribute(Contact.PUBLISHER_NAME);
+
+        if (pocName != null && StringUtils.isNotEmpty((String) pocName.getValue())) {
```

Comment:
I'm not crazy about casting pocName.getValue() to a String without first testing to make sure it's safe (ie. instanceof).